src/System.Runtime.Extensions/src/System/Net/WebUtility.cs

```diff
@@ -148,27 +173,44 @@ public static string HtmlDecode(string value)
                 return value;
             }
 
-            // Don't create StringBuilder if we don't have anything to encode
             if (!StringRequiresHtmlDecoding(value))
             {
                 return value;
             }
 
-            StringBuilder sb = StringBuilderCache.Acquire(value.Length);
-            HtmlDecode(value, sb);
-            return StringBuilderCache.GetStringAndRelease(sb);
+            var sb = new ValueStringBuilder(value.Length);
+            HtmlDecode(value, ref sb);
+            return sb.ToString();
         }
 
         public static void HtmlDecode(string value, TextWriter output)
         {
-            output.Write(HtmlDecode(value));
+            if (output == null)
+            {
+                throw new ArgumentNullException(nameof(output));
+            }
+
+            if (string.IsNullOrEmpty(value))
+            {
+                output.Write(value);
```

Comment:
This will presumably throw if value is null. But, I see previously it would have also thrown doing `value.Length`.

Reply:
Seems we should probably throw ANE if value is null.

Reply:
Good spot, will do.

Reply:
I'm now throwing an `ArgumentNullException`, but I'm concerned that this is technically a breaking change (see [my comment](https://github.com/dotnet/corefx/pull/27250#issuecomment-371223676)). What do you think?

Reply:
We are generally OK with changing NRE->ANE.

Reply:
Oh I see you changed back. That's fine too.